src/Model/Response.php

```diff
@@ -84,66 +80,58 @@ public static function for400WithForm(string $formType, array $validationGroups
         );
     }
 
-    /**
-     * @param string[]|string $description
-     */
-    public static function for401($description = ['Authentication is missing, invalid or expired']): self
+    public static function for401(): self
     {
-        return new self(401, (array) $description, RFC7807ErrorOutput::for401());
+        return new self(401, ['Authentication is missing, invalid or expired'], RFC7807ErrorOutput::for401());
     }
 
-    /**
-     * @param string[]|string $description
-     */
-    public static function for403($description = ['Returned when operation is prohibited']): self
+    public static function for403(): self
     {
-        return new self(403, (array) $description, RFC7807ErrorOutput::for403());
+        return new self(403, ['Returned when operation is prohibited'], RFC7807ErrorOutput::for403());
     }
 
-    /**
-     * @param string[]|string $description
-     */
-    public static function for404($description = ['Returned when resource is not found']): self
+    public static function for404(): self
     {
-        return new self(404, (array) $description, RFC7807ErrorOutput::for404());
+        return new self(404, ['Returned when resource is not found'], RFC7807ErrorOutput::for404());
     }
 
-    /**
-     * @param string[]|string $description
-     */
-    public static function for406($description = ['Returned when response content type expected is not supported']): self
+    public static function for406(): self
     {
         return new self(
             406,
-            (array) $description,
+            ['Returned when response content type expected is not supported'],
             RFC7807ErrorOutput::for406()
         );
     }
 
-    /**
-     * @param string[]|string $description
-     */
-    public static function for415($description = ['Returned when request payload format is not supported']): self
+    public static function for415(): self
     {
         return new self(
             415,
-            (array) $description,
+            ['Returned when request payload format is not supported'],
             RFC7807ErrorOutput::for415()
         );
     }
 
-    /**
-     * @param string[]|string $description
-     */
-    public static function for500($description = ['Returned on server error']): self
+    public static function for500(): self
     {
         return new self(
             500,
-            (array) $description,
+            ['Returned on server error'],
             RFC7807ErrorOutput::for500()
         );
     }
 
+    /**
+     * @param string[]|string $description
+     */
+    public function withDescription($description): self
```

Comment:
Shouldn't we allow string or array ?

Reply:
Previously we allowed description to be array or string so I decided to keep this approach.
Maybe we should have 2 methods:

`public function withDescription(array $description): self`
`public function withDescriptionText(string $description): self`

What do you think?

Reply:
i think allowing both is good for now